shared/src/main/kotlin/org/javacs/kt/classpath/Home.kt

```diff
@@ -1,18 +1,26 @@
 package org.javacs.kt.classpath
 
-import org.javacs.kt.util.userHome
+import java.nio.file.Files
+import java.nio.file.Path
 import java.nio.file.Paths
+import org.javacs.kt.util.KotlinLSException
+import org.javacs.kt.util.userHome
 
+private fun createPathOrNull(envVar: String): Path? = System.getenv(envVar)?.let(Paths::get)
 
-internal val gradleHome =
-    System.getenv("GRADLE_USER_HOME")?.let { Paths.get(it) }
-        ?: userHome.resolve(".gradle")
-
-internal val mavenHome =
-    System.getenv("MAVEN_HOME")?.let { Paths.get(it) }
-        ?: System.getenv("M2_HOME")?.let { Paths.get(it) }
-        ?: userHome.resolve(".m2")
+private val possibleMavenRepositoryPaths =
+    sequenceOf(
+        createPathOrNull("MAVEN_REPOSITORY"),
+        createPathOrNull("MAVEN_HOME")?.let { it.resolve("repository") },
+        createPathOrNull("M2_HOME")?.let { it.resolve("repository") },
+        userHome.resolve(".m2/repository"),
+    )
+    .filterNotNull()
 
 internal val mavenRepository =
-    System.getenv("MAVEN_REPOSITORY")?.let { Paths.get(it) }
-        ?: mavenHome.resolve("repository")
+    possibleMavenRepositoryPaths.firstOrNull { Files.exists(it) }
+        ?: throw KotlinLSException(
+            "No repositories found at \$MAVEN_REPOSITORY, \$MAVEN_HOME, \$M2_HOME or \$HOME/.m2"
+        )
```

Comment:
I think the exception might have broken all clients that didn't have a Maven repository (e.g. #564). We should really be a bit more careful with silently changing the type of something, this had to be optional for the class path resolver logic to work properly...

Reply:
Won't reverting this change silently fail in other ways now because it will set a maven repository that doesn't actually exist? I'm not really sure where and what `mavenRepository` is used for. But if mavenRepository is optional, then we can refactor the code to make it so. Instead of setting a value that is wrong.

I could help with this if that is a useful approach. Currently I am running into the same issue I referenced [here](https://github.com/fwcd/kotlin-language-server/issues/430#issuecomment-1750054676) setting up Kotlin on a new machine.